### PR TITLE
#146 EA-1811 updated entity grouping with reset the value of the total field

### PR DESCRIPTION
**Pull request by rgraf**
Adjusted create, update and delete methods.

### CODE REVIEW

File: entity-web/src/main/java/eu/europeana/entity/web/controller/ConceptSchemeController.java [hunk 1]
```diff
@@ -242,7 +245,9 @@ public void setEntityService(EntityService entityService) {
 	    } else {
 		httpStatus = HttpStatus.NO_CONTENT;
 		ConceptScheme updated = getEntityService().disableConceptScheme(existingConceptScheme);
-		eTag = generateETag(updated.getModified(), null);
+		//update entities with concept scheme
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/controller/ConceptSchemeController.java]: please update the comment according to the code which is removing the concept scheme from entities, not updating

File: entity-web/src/main/java/eu/europeana/entity/web/service/impl/EntityServiceImpl.java [hunk 2]
```diff
@@ -159,26 +157,34 @@ public String resolveByUri(String uri) throws HttpException {
...
 	} else {// suggest
 
-	    // ConceptScheme Not Supported in suggester
-	    if (entityTypes.contains(EntityTypes.ConceptScheme))
-		throw new ParamValidationException(I18nConstants.INVALID_PARAM_VALUE,
-			WebEntityConstants.QUERY_PARAM_TYPE, EntityTypes.ConceptScheme.getInternalType());
-
-	    if (entityTypes.contains(EntityTypes.All)) {
+	    if (isEmptyOrAll(entityTypes)) {
 		entityTypes.clear();
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/service/impl/EntityServiceImpl.java]: We can still get a NPE at this stage, please initialize the list if null

File: entity-web/src/main/java/eu/europeana/entity/web/service/impl/EntityServiceImpl.java [hunk 3]
```diff
@@ -557,20 +561,20 @@ public ConceptScheme updateEntitiesWithConceptScheme(ConceptScheme storedConcept
...
+//	String type = parameters.getFirst(WebEntityConstants.QUERY_PARAM_TYPE);
+//
+//	if (StringUtils.isBlank(type))
+//	    type = EntityTypes.All.name();
+//	else
+//	    type = URLDecoder.decode(type, StandardCharsets.UTF_8.name());
+//
+//	entityTypes = getEntityTypesFromString(type);
+//	return entityTypes;
+//    }
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/service/impl/EntityServiceImpl.java]: please remove the commented out code

File: entity-web/src/main/java/eu/europeana/entity/web/controller/ConceptSchemeController.java (hunk 1 above)
Comment (rgraf) [reply entity-web/src/main/java/eu/europeana/entity/web/controller/ConceptSchemeController.java]: Comment changed

File: entity-web/src/main/java/eu/europeana/entity/web/service/impl/EntityServiceImpl.java (hunk 2 above)
Comment (rgraf) [reply entity-web/src/main/java/eu/europeana/entity/web/service/impl/EntityServiceImpl.java]: Initialized list if it is null before clear

File: entity-web/src/main/java/eu/europeana/entity/web/service/impl/EntityServiceImpl.java (hunk 3 above)
Comment (rgraf) [reply entity-web/src/main/java/eu/europeana/entity/web/service/impl/EntityServiceImpl.java]: Comments removed

Comment (gsergiu) [reply entity-web/src/main/java/eu/europeana/entity/web/service/impl/EntityServiceImpl.java]: you commited a merge conflict ... please fix it

Comment (rgraf) [reply entity-web/src/main/java/eu/europeana/entity/web/service/impl/EntityServiceImpl.java]: fixed

**Pull request merged**